luigi/contrib/s3.py

```diff
@@ -553,16 +416,57 @@ def mkdir(self, path, parents=True, raise_if_exists=False):
         if raise_if_exists and self.isdir(path):
             raise FileAlreadyExists()
 
-        _, key = self._path_to_bucket_and_key(path)
+        bucket, key = self._path_to_bucket_and_key(path)
         if self._is_root(key):
-            return  # isdir raises if the bucket doesn't exist; nothing to do here.
+            # isdir raises if the bucket doesn't exist; nothing to do here.
+            return
 
-        key = self._add_path_delimiter(key)
+        path = self._add_path_delimiter(path)
 
-        if not parents and not self.isdir(os.path.dirname(key)):
+        if not parents and not self.isdir(os.path.dirname(path)):
             raise MissingParentDirectory()
 
-        return self.put_string("", self._add_path_delimiter(path))
+        return self.put_string("", path)
+
+    def listdir(self, path, start_time=None, end_time=None, return_key=False):
+        """
+        Get an iterable with S3 folder contents.
+        Iterable contains paths relative to queried path.
+        :param start_time: Optional argument to list files with modified datetime after start_time
+        :param end_time: Optional argument to list files with modified datetime before end_time
+        :param return_key: Optional argument, when set to True will return boto3's ObjectSummary (instead of the filename)
+        """
+        (bucket, key) = self._path_to_bucket_and_key(path)
+
+        # grab and validate the bucket
+        s3_bucket = self.s3.Bucket(bucket)
+
+        key_path = self._add_path_delimiter(key)
+        key_path_len = len(key_path)
+        for item in s3_bucket.objects.filter(Prefix=key_path):
+            last_modified_date = item.last_modified
```

Comment:
Mostly a note for myself - and maybe worth updating the docstring... `last_modified_date` is now a datetime with timezone. So `start_time`/`end_time` must now also be offset-aware datetimes (or `None`).